src/policy.c

```diff
@@ -522,6 +563,16 @@ policy_auto_assign_new_device(device_t *device)
     res = usbowls_plug_device(vm->domid, device->busid, device->devid);
     if (res != 0)
       device->vm = NULL;
+    xd_log(LOG_INFO,
+        "Automatically assigned device [Bus=%03d, Dev=%03d, VID=%04X, PID=%04X, Serial=%s] to VM [UUID=%s, DomID=%d], according to policy rule %d",
+        device->busid,
+        device->devid,
+        device->vendorid,
+        device->deviceid,
+        device->serial,
+        vm->uuid,
+        vm->domid,
+        rule->pos);
```

Comment:
SEGV here, rule can be `NULL`.

Triggers on [$10131](Build [$10131](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10131) when plugging a USB thumbdrive when a VM is focused.
```
#0  policy_auto_assign_new_device (device=device@entry=0x428e80) at ../../git/src/policy.c:566
#1  0x0000000000406177 in udev_maybe_add_device (dev=dev@entry=0x42d420, auto_assign=auto_assign@entry=1) at ../../git/src/udev.c:494
#2  0x0000000000406775 in udev_event () at ../../git/src/udev.c:617
#3  0x0000000000403acf in main () at ../../git/src/main.c:135
p rule
$1 = (rule_t *) 0x0
```

`default_lookup` can fail and return `NULL`, maybe that's not intended?

Reply:
Ah, it's not that default_lookup is failing, it's that the policy lookup after doesn't track the rule used for making the decision. Added a commit to add a parameter to policy_is_allowed to be able to keep track of the rule used to make the policy decision. Built and tested that it addresses the vm-focus auto-assign.

Reply:
Ah I misread that.

`policy_is_allowed` now has a more complex behavior:
- Allow rule: Returns `true` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Deny rule: Returns `false` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Default Deny catch-all: Returns `false` without touching `*rules_ptr`;